### ISSUE TITLE
Provide functions in addition to macros

### ISSUE THREAD

**Issue by hogandp** [ndortega/Oxygen.jl#91]
Please expose functions in addition to your macros for @get, @put, .... @staticfiles.

The general use case of having a globally defined HTTP_ROOT that gets fed to @staticfiles is broken e.g.

```julia
using Oxygen

HTTP_ROOT = "/some/default/path"

# These works but really isn't that helpful
@staticfiles "/some/default/path" "static"

# This does not work
@staticfiles HTTP_ROOT "static"

function start(httpRoot::String = HTTP_ROOT)
    # This does not work
    @staticfiles httpRoot "static"
    
    # This also does not work
    @staticfiles(httpRoot, "static")    
end
```

**Comment by ndortega** [comment]
Hi @hogandp , 

What's your use case that requires functions rather than macros? Is it more for stylistic reasons or are you trying to do something else that macros can't support or handle well?

I could take a look into providing functions instead of macros, but the syntax naturally would have to change 
to look something like the following:

```julia

get("/point/{x::String}/{y::Int}") do (request, x, y)
    println("new point! ($x, $y)")
end
# OR
get("/point/{x}/{y}") do (request, x, y)::Tuple{HTTP.Request, String, Int}
    println("new point! ($x, $y)")
end
# OR
get("/point/{x}/{y}", function(request, x::Int, y::Int)
    println("new point! ($x, $y)")
end) 
```

The @staticfiles macro was designed to "mount" a folder only on startup and must be passed a string and not a variable. Granted, we could just escape the incoming path so that we could pass variables and handle string interpolation. 

Just so I get a better idea of your use case, are you looking to mount other files/folders at runtime?

**Comment by ndortega** [comment]
Hi @hogandp 

1.) There's a function called `internalrequest` which you can use to hit your endpoints without having to start the server. I make use of this to handle 90% of my unit tests for this library.

Here's an example:
```julia
internalrequest(HTTP.Request("GET", "/test"))
```

About your second point, you can pass variables to all routing macros.  I have an example of how to dynamically register routes with the server: https://github.com/ndortega/Oxygen.jl#interpolating-variables-into-endpoints

2.) Absolutely, I can make this more dynamic by escaping the parameters so your example will work. Honestly, these could even just be plain old functions since the macro syntax isn't doing anything super useful

Let me know if I addressed all your points!

**Comment by hogandp** [comment]
Sounds good to me!

**Comment by ndortega** [comment]
@hogandp 

I just released version 1.1.9 which officially includes support for routing functions. Check it out and let me know what you think!
